### PR TITLE
osx: restore code for ~/keybase mountpoint and helper dialog prompt

### PR DESCRIPTION
**Pull request by strib**
This PR consists mostly of code that was reverted in #10564, including:

* Moving the OSX mountpoint to `~/keybase`.
* Popping up a dialog box before asking for a password when installing the new helper.
screen shot 2018-02-22 at 4 26 57 pm

* Better handling of hitting "cancel" when asked for helper password (uninstalls fuse in that case).

Of course, instead of making a symlink from `/keybase`, this new code just mounts the redirector there.

I think the only interesting thing to review here is the dialog text, which may change slightly if we decide to move the mount out of the home directory.

Issue: KBFS-2782

### CODE REVIEW

File: osx/KBKit/KBKit/System/KBEnvConfig.m [hunk 1]
```diff
@@ -36,26 +37,44 @@ + (instancetype)envConfigWithRunMode:(KBRunMode)runMode {
...
+          mountDir = (NSString *)obj;
+        }
+      }
+    }
+
     switch (_runMode) {
       case KBRunModeProd: {
         self.title = @"Keybase.io";
-        self.mountDir = [KBPath path:@"/keybase" options:0];
+        self.mountDir = mountDir;
```
Comment (songgao) [inline osx/KBKit/KBKit/System/KBEnvConfig.m]: This would be `~/keybase` now?

Comment (strib) [reply osx/KBKit/KBKit/System/KBEnvConfig.m]: Yep, assuming nothing is set in config.json.

**Comment by strib** [comment]
> LGTM; For text, what do you think either

`keybase/` was from chris's original text, so I didn't want to change it.  I think he wanted to avoid `~` because not everyone knows what that is.  I think I'm going to wait to see what our final decision is on the mount location before deciding on a change (since putting it in the home directory messes up backup programs and `du`).

File: osx/KBKit/KBKit/Component/KBHelperTool.m [hunk 2]
```diff
@@ -96,6 +116,7 @@ - (void)refreshComponent:(KBRefreshComponentCompletion)completion {
       if ([bundleVersion isGreaterThan:runningVersion]) {
         if (bundleVersion) info[@"Bundle Version"] = [bundleVersion description];
         self.componentStatus = [KBComponentStatus componentStatusWithInstallStatus:KBRInstallStatusInstalled installAction:KBRInstallActionUpgrade info:info error:nil];
+        [self doInstallAlert:bundleVersion runningVersion:runningVersion];
```
Comment (jzila) [inline osx/KBKit/KBKit/Component/KBHelperTool.m]: Is this correct? `doInstallAlert: bundleVersion`?

File: go/client/cmd_install_osx.go [hunk 3]
```diff
@@ -154,6 +154,9 @@ func exitOnError(result keybase1.InstallResult) {
 		os.Exit(1)
 	}
 	for _, r := range result.ComponentResults {
+		if r.ExitCode != 0 {
+			os.Exit(r.ExitCode)
```
[the remark this answers is not in the document]
Comment (strib) [reply go/client/cmd_install_osx.go]: All the statuses are already logged in `Run()`, before this function is called (if the json format is enabled).  This PR doesn't change any of the logic associated with that, just makes the exit code better.

File: osx/KBKit/KBKit/Component/KBHelperTool.m (hunk 2 above)
Comment (strib) [reply osx/KBKit/KBKit/Component/KBHelperTool.m]: It compiles and runs correctly. What's the concern?  You're wondering if there should be a space after the `:`?  I guess with objc, the first param can go unnamed and directly after the function name, but the other params need there names.  Or something.  Just following the other calls in the rest of the code.

Comment (jzila) [reply osx/KBKit/KBKit/Component/KBHelperTool.m]: I was just confused about what was being assigned: `doInstallAlert` and `bundleVersion` don't seem to have much to do with one another.

Comment (jzila) [reply osx/KBKit/KBKit/Component/KBHelperTool.m]: Unless it just means "do the install with an alert"

Comment (strib) [reply osx/KBKit/KBKit/Component/KBHelperTool.m]: It's a function call, calling `doInstallAlert` with 2 params, `bundleVersion` and `runningVersion`.  The function pops up an alert about the newly installed version, depending on what the two params are.

Comment (jzila) [reply osx/KBKit/KBKit/Component/KBHelperTool.m]: Oh yeah. God it's so hard to read with some params being named and some not being named.

**Pull request merged**